### PR TITLE
Support RGBA for quadmesh mode of pcolorfast.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

As pointed out by @ImportanceOfBeingErnest in https://stackoverflow.com/questions/49621655/rgb-polar-plot-in-python/49630831#49630831, pcolormesh does support (in a slightly awkward way) RGBA input, and thus pcolorfast can also support it easily.

Suggesting to still smuggle this in 3.1, as it's a natural extension over https://github.com/matplotlib/matplotlib/pull/8690...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by efiring** [comment]
I like this and will give it a positive review ASAP, but I request that you first add a fix for a relevant documentation error that was introduced in 3e14dd2ce.  Specifically, in the Quadmesh case, the shapes of X and Y *must* be (M+1, N+1), where C.shape is (M, N).  Also remove the reference to pcolormesh, which is misleading in this context; the shape rules are more strict for pcolorfast than for pcolormesh.

**Comment by anntzer** [comment]
edited the docstring accordingly.

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -6331,7 +6339,7 @@ def pcolorfast(self, *args, alpha=None, norm=None, cmap=None, vmin=None,
 
         if vmin is not None or vmax is not None:
             ret.set_clim(vmin, vmax)
-        else:
+        elif np.ndim(C) == 2:
             ret.autoscale_None()
```
Comment (timhoffm) [inline lib/matplotlib/axes/_axes.py]: I would add a comment
~~~
# np.dim(C) == 3 has explicit RGB(A) values and does not need scaling.
~~~

Comment (anntzer) [reply lib/matplotlib/axes/_axes.py]: sure, done

Comment (timhoffm) [reply lib/matplotlib/axes/_axes.py]: Have you pushed? I don't see a comment.

Comment (anntzer) [reply lib/matplotlib/axes/_axes.py]: duh, put it at the wrong line; fixed.

**Comment by QuLogic** [comment]
Well, the rc is out, so I'm guessing we aren't going to slip this into 3.1.0.

**Pull request merged**